sympy/simplify/tests/test_simplify.py

```diff
@@ -297,6 +297,14 @@ def test_separatevars():
     eq = x*(1 + hyper((), (), y*z))
     assert separatevars(eq) == eq
 
+    s = separatevars(abs(x*y))
+    assert s == abs(x)*abs(y) and s.is_Mul
+    z = cos(1)**2 + sin(1)**2 - 1
+    a = abs(x*z)
+    assert separatevars(a) == a
```

Comment:
Is this to test that z is factored out automatically and that separatevars does not put it back in or the opposite?
```suggestion
    assert separatevars(a) == a == abs(x)*abs(z)
```
or `!=`.

Reply:
Good sighting. This is a question and is perhaps the source of the simplification problem with `abs(5*(x**2-1)/(x+1))`. If it's univariate, `separatevars` doesn't do anything. But in case of the the `abs`, perhaps something should be done.

Reply:
OK! (I updated the test from `=` to `==`.) No idea really.

I do not really grasp the complete simplify process, but it still seems to me that a separate `simplify_abs` that applies `collect_abs` (or something very similar), do simplify on the arguments and then evaluates as is would solve this and maybe some other problems.